python/cuml/metrics/confusion_matrix.py

```diff
@@ -0,0 +1,114 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import numpy as np
+import cupy as cp
+
+from cuml.utils import input_to_cuml_array
+from cuml.utils.memory_utils import with_cupy_rmm
+from cuml.metrics.utils import sorted_unique_labels
+from cuml.prims.label import make_monotonic
+
+
+@with_cupy_rmm
+def confusion_matrix(y_true, y_pred,
+                     labels=None,
+                     sample_weight=None,
+                     normalize=None):
+    """Compute confusion matrix to evaluate the accuracy of a classification.
+
+    Parameters
+    ----------
+    y_true : array-like (device or host) shape = (n_samples,)
+        or (n_samples, n_outputs)
+        Ground truth (correct) target values.
+    y_pred : array-like (device or host) shape = (n_samples,)
+        or (n_samples, n_outputs)
+        Estimated target values.
+    labels : array-like (device or host) shape = (n_classes,), optional
+        List of labels to index the matrix. This may be used to reorder or
+        select a subset of labels. If None is given, those that appear at least
+        once in y_true or y_pred are used in sorted order.
+    sample_weight : array-like (device or host) shape = (n_samples,), optional
+        Sample weights.
+    normalize : string in [‘true’, ‘pred’, ‘all’]
+        Normalizes confusion matrix over the true (rows), predicted (columns)
+        conditions or all the population. If None, confusion matrix will not be
+        normalized.
+
+    Returns
+    -------
+    C : array-like (device or host) shape = (n_classes, n_classes)
+        Confusion matrix.
+    """
+    y_true, n_rows, n_cols, dtype = \
+        input_to_cuml_array(y_true, check_dtype=[cp.int32, cp.int64])
+    y_true = y_true.to_output('cupy')
+
+    y_pred, _, _, _ = \
+        input_to_cuml_array(y_pred, check_dtype=dtype,
+                            check_rows=n_rows, check_cols=n_cols)
+    y_pred = y_pred.to_output('cupy')
+
+    if labels is None:
+        labels = sorted_unique_labels(y_true, y_pred)
+        n_labels = len(labels)
+    else:
+        labels, n_labels, _, _ = \
+            input_to_cuml_array(labels, check_dtype=dtype, check_cols=1)
+        labels = labels.to_output('cupy')
+    if sample_weight is None:
+        sample_weight = cp.ones(n_rows, dtype=dtype)
+    else:
+        sample_weight, _, _, _ = \
+            input_to_cuml_array(sample_weight,
+                                check_dtype=[cp.float32, cp.float64,
+                                             cp.int32, cp.int64],
+                                check_rows=n_rows, check_cols=n_cols)
+        sample_weight = sample_weight.to_output('cupy')
+
+    if normalize not in ['true', 'pred', 'all', None]:
+        raise ValueError("normalize must be one of {'true', 'pred', "
```

Comment:
Similar to the Dask exception, it'd be useful to print the value of `normalize` in the exception